plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1074,14 +1074,13 @@ func getMatchingAntiAffinityTerms(pod *v1.Pod, nodeInfoMap map[string]*scheduler
 				continue
 			}
 			for _, term := range getPodAntiAffinityTerms(affinity.PodAntiAffinity) {
-				namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, &term)
+				namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(existingPod, &term)
```

Comment:
this was a bug fix
A separate PR with test for it was added in https://github.com/kubernetes/kubernetes/pull/45152